.github/workflows/codeql.yaml

```
@@ -0,0 +1,46 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    env:
+      SIGNING_KEY: ${{ secrets.SIGNING_SECRET_KEY_RING_FILE }}
+      NEXUS_UNAME: ${{ secrets.NEXUS_USERNAME }}
+      NEXUS_PWD: ${{ secrets.NEXUS_PASSWORD }}
+      SIGNING_KEYID: ${{ secrets.SIGNING_KEY_ID }}
+      SIGNING_PWD: ${{ secrets.SIGNING_PASSWORD }}
```

I think I see some unused secrets, remove the one that are not being used.
```suggestion
    env:
      NEXUS_UNAME: ${{ secrets.NEXUS_USERNAME }}
      NEXUS_PWD: ${{ secrets.NEXUS_PASSWORD }}
```